src/app/footer/footer.ts

```diff
@@ -0,0 +1,56 @@
+import {Component} from 'angular2/core';
+import {Location} from 'angular2/router';
+
+@Component({
+  selector: 'publish-footer',
+  styleUrls: ['app/footer/footer.css'],
+  template: `
```

Comment:
My 2c: At this point I am starting to think that inline is almost always better with the caveat that current tools don't give you syntax highlighting (which I expect to change).